src/rest.js

```diff
@@ -120,101 +118,109 @@ export class Rest {
    */
-  patchOne(resource: string, id: string|Number, criteria?: {}, body?: {}, options?: {}): Promise<any|Error> {
+  patchOne(resource: string, id: string|Number, criteria?: {}, body?: any, options?: any): Promise<any|Error> {
     return this.request('PATCH', getRequestPath(resource, id, criteria), body, options);
   }
 
   /**
    * Delete a resource.
    *
-   * @param {string}           resource  The resource to delete
-   * @param {{}|string|Number} criteria  Object for where clause, string / number for id.
-   * @param {{}}               [options] Extra fetch options.
+   * @param {string}           resource   The resource to delete
+   * @param {{}|string|Number} [criteria] Object for where clause, string / number for id.
+   * @param {{}}               [options]  Extra request options.
    *
    * @return {Promise<any>|Promise<Error>} Server response as Object
    */
-  destroy(resource: string, criteria?: {}|string|Number, options?: {}): Promise<any|Error> {
+  destroy(resource: string, criteria?: {}|string|Number, options?: any): Promise<any|Error> {
     return this.request('DELETE', getRequestPath(resource, criteria), undefined, options);
   }
 
   /**
    * Delete a resource.
    *
-   * @param {string}           resource  The resource to delete
-   * @param {string|Number}    id        String / number for id to be added to the path.
-   * @param {{}}               criteria  Object for where clause
-   * @param {{}}               [options] Extra fetch options.
+   * @param {string}           resource   The resource to delete
+   * @param {string|Number}    id         String / number for id to be added to the path.
+   * @param {{}}               [criteria] Object for where clause
+   * @param {{}}               [options]  Extra request options.
    *
    * @return {Promise<any>|Promise<Error>} Server response as Object
    */
-  destroyOne(resource: string, id: string|Number, criteria?: {}, options?: {}): Promise<any|Error> {
+  destroyOne(resource: string, id: string|Number, criteria?: {}, options?: any): Promise<any|Error> {
     return this.request('DELETE', getRequestPath(resource, id, criteria), undefined, options);
   }
 
   /**
    * Create a new instance for resource.
    *
-   * @param {string} resource  The resource to create
-   * @param {{}}     body      The data to post (as Object)
-   * @param {{}}     [options] Extra fetch options.
+   * @param {string}           resource  The resource to create
+   * @param {{}}               [body]    The data to post (as Object)
+   * @param {{}}               [options] Extra request options.
    *
-   * @return {Promise<any>|Promise<Error>} Server response as Object
+   * @return {Promise<any>} Server response as Object
    */
-  create(resource: string, body?: {}, options?: {}): Promise<any|Error> {
-    return this.post(...arguments);
+  create(resource: string, body?: any, options?: any): Promise<any|Error> {
```

Comment:
main thing was that linter doesn't like arguments. ...rest should be used. but then create would loose it argument types. hence passing them all on as solution. the any was, since my post required any there now. but i might better change those options to {} or object everywhere